### PR TITLE
add View > Files

### PR DESCRIPTION
**Pull request by ddennedy**
TODO

- [x] Locations bar at the top (similar to Presets)
- [x] configurable Open With sub-menu, separate list per-media type, not extension or mime-type
- [x] Show in File Manager
- [x] Properties > Show in Files

### CODE REVIEW

**Review by bmatherly — COMMENTED** [review]
I was expecting more media information to be displayed (resolution, frame rate, etc). Is that planned for a future enhancement?

File: src/docks/filesdock.cpp [hunk 1]
```diff
@@ -0,0 +1,1267 @@
...
+                QStringLiteral(""),
+            };
+            auto i = isDir ? 4 : mediaType(index);
+            return names[i];
+        }
+        switch (role) {
+        case Qt::ToolTipRole:
+            return QDir::toNativeSeparators(filePath(index));
+        case DateRole:
+            return fileInfo(index).lastModified();
```
[the remark this answers is not in the document]
Comment (ddennedy) [reply src/docks/filesdock.cpp:263]: Any more metadata than this requires more interrogation and a database of some kind. It is intentionally not currently trying to load every file into MLT as that was killing performance, which is already not great. Imagine trying to multiple open MLT XML projects in a full directory. As a result it uses heuristics and only loads files for certain things under certain circumstances. For first version it is only an integrated simple file browser.

[the remark this answers is not in the document]
Comment (bmatherly) [reply src/docks/filesdock.cpp:263]: I notice that the FilesMediaTypeTask is already loading each file in a Producer.
https://github.com/mltframework/shotcut/blob/0798fd4cd75f6c70d411595dc1cf148f08e6443a/src/docks/filesdock.cpp#L150

So I thought that maybe 90% of the work is already being done. But I understand that for some file types it is also required to decode some frames. And there would also be extra complication of creating a structured database to store the metadata in.

**Comment by bmatherly** [comment]
This feature is good to go from my perspective. Nice work.

File: src/docks/filesdock.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ddennedy) [reply src/docks/filesdock.cpp:262]: Not every file; see `mediaType()`. There is a cache plus file extension heuristic:

https://github.com/mltframework/shotcut/blob/0798fd4cd75f6c70d411595dc1cf148f08e6443a/src/docks/filesdock.cpp#L306

When you consider all files on the file system, most will get a producer. But if the file is not media; it is quickly rejected. It is slower when there is valid media, which is why I added the file extension heuristic.

**Pull request merged**